### PR TITLE
Implement SSS sets

### PR DESCRIPTION
**Pull request by jdent02**

### CODE REVIEW

File: operators/__init__.py [hunk 1]
```diff
@@ -74,6 +74,51 @@ def invoke(self, context, event):
         return {'FINISHED'}
 
 
+class AppleseedAddSssSet(bpy.types.Operator):
+    """
+    Operator for adding material layers.
```
Comment (dictoon) [inline operators/__init__.py]: Fix comment.

File: operators/__init__.py [hunk 2]
```diff
@@ -74,6 +74,51 @@ def invoke(self, context, event):
...
+        collection.add()
+        num = collection.__len__()
+        collection[num - 1].name = "SSS Set " + str(num)
+
+        return {'FINISHED'}
+
+
+class AppleseedRemoveSssSet(bpy.types.Operator):
+    """
+    Operator for removing material layers.
```
Comment (dictoon) [inline operators/__init__.py]: Fix comment.

File: projectwriter.py [hunk 3]
```diff
@@ -654,6 +654,8 @@ def __emit_object_instance_element(self, object_name, instance_name, instance_ma
         if object.appleseed.ray_bias_method != 'none':
             self.__emit_parameter("ray_bias_method", object.appleseed.ray_bias_method)
             self.__emit_parameter("ray_bias_distance", object.appleseed.ray_bias_distance)
+        if object.appleseed.object_sss_set != "":
+            self.__emit_parameter("sss_set_id", object.appleseed.object_sss_set.replace(" ", "_"))
```
Comment (dictoon) [inline projectwriter.py]: Is there a particular reason behind doing these replacements? SSS sets are arbitrary strings in appleseed, in particular spaces are allowed.

File: properties/world.py [hunk 4]
```diff
@@ -29,6 +29,28 @@
 import bpy
 
 
+class AppleseedSSSSetsProps(bpy.types.PropertyGroup):
+    name = bpy.props.StringProperty(name="SSS Set Name",
+                                        default="")
```
Comment (dictoon) [inline properties/world.py]: Make a single line.

File: properties/world.py [hunk 5]
```diff
@@ -29,6 +29,28 @@
...
+
+
+class AppleseedSSSSets(bpy.types.PropertyGroup):
+
+    @classmethod
+    def register(cls):
+        bpy.types.Scene.appleseed_sss_sets = bpy.props.PointerProperty(name="appleseed_sss_sets",
+                                                                  description="appleseed SSS Sets",
```
Comment (dictoon) [inline properties/world.py]: Fix indentation in this class.

File: ui/world.py [hunk 6]
```diff
@@ -89,9 +89,50 @@ def draw(self, context):
...
+    bl_label = 'SSS Sets'
+    bl_space_type = "PROPERTIES"
+    bl_region_type = "WINDOW"
+    bl_context = "world"
+    COMPAT_ENGINES = {'APPLESEED_RENDER'}
+
+    @classmethod
+    def poll(cls, context):
+        renderer = context.scene.render
+        return renderer.engine == 'APPLESEED_RENDER' and context.scene is not None
```
Comment (dictoon) [inline ui/world.py]: Isn't `COMPAT_ENGINES = {'APPLESEED_RENDER'}` enough?

File: ui/world.py [hunk 7]
```diff
@@ -89,9 +89,50 @@ def draw(self, context):
...
+
+    def draw(self, context):
+        layout = self.layout
+        scene = context.scene
+        world = scene.appleseed_sss_sets
+
+        row = layout.row()
+        row.template_list("SSS_sets_slots", "", world,
+                              "sss_sets", world, "sss_sets_index", rows=1, maxrows=16, type="DEFAULT")
```
Comment (dictoon) [inline ui/world.py]: Fix indentation.

File: ui/world.py [hunk 8]
```diff
@@ -89,9 +89,50 @@ def draw(self, context):
...
+
+        row = layout.row(align=True)
+        row.operator("appleseed.add_sss_set", text="Add Set", icon="ZOOMIN")
+        row.operator("appleseed.remove_sss_set", text="Remove Set", icon="ZOOMOUT")
+
+        if world.sss_sets:
+                current_set = world.sss_sets[world.sss_sets_index]
```
Comment (dictoon) [inline ui/world.py]: Fix indentation.

File: ui/world.py [hunk 9]
```diff
@@ -89,9 +89,50 @@ def draw(self, context):
...
+                layout.prop(current_set, "name", text="SSS Set Name")
+
```
Comment (dictoon) [inline ui/world.py]: Remove extra blank lines.

File: operators/__init__.py [hunk 10]
```diff
@@ -74,6 +74,47 @@ def invoke(self, context, event):
         return {'FINISHED'}
 
 
+class AppleseedAddSssSet(bpy.types.Operator):
+    """Operator for adding material layers"""
```
Comment (dictoon) [inline operators/__init__.py]: The comments here and below are still incorrect, it seems.

Comment (jdent02) [reply operators/__init__.py]: Ah crap. You are correct

**Comment by jdent02** [comment]
Should be good now

**Comment by dictoon** [comment]
Thank you!

**Pull request merged**